### PR TITLE
Fix all tests that insert objects into db instead of using genesis.

### PR DESCRIPTION
**Pull request by mystenmark**
Had to do this for #8556 and would like to get it merged earlier.

(@williampsmith did most of the work on this, i just pulled the changes out into a new PR)

### CODE REVIEW

File: crates/sui/tests/shared_objects_tests.rs [hunk 1]
```diff
@@ -143,9 +156,9 @@ async fn call_shared_object_contract() {
         .unwrap();
     let increment_transaction = *effects.transaction_digest();
     assert!(matches!(effects.status(), ExecutionStatus::Success { .. }));
-    // Again - only gas object transaction and counter creation are dependencies
+    // Again - only genesis, increment, and counter creation are dependencies
     // Previously executed assert_value transaction(s) are not a dependency because they took immutable reference to shared object
-    assert_eq!(effects.dependencies().len(), 2);
+    assert_eq!(effects.dependencies().len(), 3);
```
Comment (williampsmith) [inline crates/sui/tests/shared_objects_tests.rs:161]: Where is this change coming from?

**Review by williampsmith — APPROVED** [review]
LGTM assuming it passes CI

**Review by bmwill — COMMENTED** [review]
Can we completely remove the function that does this side loading of objects into the db now?

**Comment by mystenmark** [comment]
> Can we completely remove the function that does this side loading of objects into the db now?

I will try

**Review by sadhansood — APPROVED** [review]

**Pull request merged**